index.js

```diff
@@ -78,5 +83,162 @@ function loader (mcVersion) {
+    }
+    /**
+     *
+     * @param {Item} itemOne left hand item
+     * @param {Item} itemTwo right hand item
+     * @returns {[xpLevelCost, fixedDurability, usedMats]}
+     * xpLevelCost is the number of xp levels used for repair (if any)
+     * fixedDurability is duribility after using the anvil
+     * usedMats is the number of materials used to fix the broken item (if many mats is used)
+     */
+    function repairCost (itemOne, itemTwo) {
+      const MAX_DURABILITY = maxItemDura[itemOne.name]
+      const { metadata: DURABILITY_LOST, displayName: toolToBeFixed } = itemOne
+      const { displayName: fixMaterial } = itemTwo
+      const [, fixMaterials] = canFixData.find(([items]) => items.includes(toolToBeFixed)) // can always be fixed by another of itself
+      if (!fixMaterials.includes(fixMaterial) && toolToBeFixed !== fixMaterial) {
+        return 0 // Enchanted book can't fix
+      }
+      if (toolToBeFixed === 'Enchanted Book') return 0 // not fixable
+      // result vars
+      let fixedDurability = 0
+      let xpLevelCost = 0
+      let usedMats = 0
+      if (itemTwo.displayName === itemOne.displayName) {
+        fixedDurability = (0.12 * MAX_DURABILITY) + itemTwo.metadata
+        xpLevelCost = 2
+        usedMats = 1
+      } else if (DURABILITY_LOST !== 0) {
+        const durabilityFixedPerMat = Math.floor(MAX_DURABILITY * 0.25)
+        const matsToFullyRepair = Math.ceil(DURABILITY_LOST / durabilityFixedPerMat)
+        if (itemTwo.count > matsToFullyRepair) { // takeall of itemTwo
+          fixedDurability = MAX_DURABILITY
+          xpLevelCost = matsToFullyRepair // 1 exp lvl per mat used
+          usedMats = matsToFullyRepair
+        } else {
+          fixedDurability = DURABILITY_LOST + (durabilityFixedPerMat * itemTwo.count)
+          xpLevelCost = itemTwo.count // 1 exp lvl per mat used
+          usedMats = itemTwo.count
+        }
+      }
+      return [xpLevelCost, fixedDurability, usedMats]
+    }
+
+    /**
+     *
+     * @param {Item} itemOne
+     * @param {Item} itemTwo
+     * @param {boolean} rename
+     */
+    function combine (itemOne, itemTwo, creative, rename) {
+      try {
+        combinePossible(itemOne, itemTwo) // throws if not possible
+        let cost = baseCost(itemOne) + baseCost(itemTwo)
+        if (rename) cost += renameCost(itemOne)
+        if (itemOne.metadata !== 0) {
+          cost += repairCost(itemOne, itemTwo)[0]
+        }
+        if (itemTwo.displayName === itemOne.displayName || itemTwo.displayName === 'Enchanted Book') {
+          const enchantCost = combineEnchants(itemOne, itemTwo, creative)[0]
+          if (enchantCost === 0 && !rename && itemOne.metadata === 0) throw new Error('No change')
+          cost += enchantCost
+        }
+        return cost
+      } catch (err) {
+        return 0 // errors should be handled by returning 0
+      }
+    }
+    function getEnchants (item) {
```

Comment:
This could be made into a method of item so it can be reused in other places.

Reply:
I just moved it into Item's scope, should it output normalized enchants?

Reply:
done